src/map/lua/lua_baseentity.cpp

```diff
@@ -2668,17 +2668,22 @@ void CLuaBaseEntity::setPos(sol::variadic_args va)
                 return;
             }
 
-            ((CCharEntity*)m_PBaseEntity)->loc.destination = zoneid;
-            ((CCharEntity*)m_PBaseEntity)->status          = STATUS_TYPE::DISAPPEAR;
-            ((CCharEntity*)m_PBaseEntity)->loc.boundary    = 0;
-            ((CCharEntity*)m_PBaseEntity)->m_moghouseID    = 0;
-            ((CCharEntity*)m_PBaseEntity)->clearPacketList();
-            charutils::SendToZone((CCharEntity*)m_PBaseEntity, 2, zoneutils::GetZoneIPP(m_PBaseEntity->loc.destination));
-            //((CCharEntity*)m_PBaseEntity)->loc.zone->DecreaseZoneCounter(((CCharEntity*)m_PBaseEntity));
+            if (static_cast<CCharEntity*>(m_PBaseEntity)->PPet != nullptr)
```

Comment:
Should be `dynamic_cast<CPetEntity*>(m_PBaseEntity)->PPet != nullptr`
PPet will never be a character but could be a mob or a pet, and we only care if pets can zone. To the best of my knowledge this applies to all "real" pets, e.g. avatar/wyvern/automaton but unsure about call beast. I can test this real quick and will report back on if we need to check pet type for jug pets

dynamic_cast is done at runtime, which is important if there's a distinction between pets and mobs like this instance

Reply:
Wait, my brain isn't working from just getting home. I think Clay's comment is right.